### PR TITLE
iasmdmd: replace init_optab() with internal checks

### PR DESCRIPTION
**Pull request by ljmf00**
init_optab() doesn't need to run everytime the compiler initiates, only
    internally on compilation/testing.

    Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

File: src/dmd/backend/iasm.d [hunk 1]
```diff
@@ -499,7 +499,7 @@ union PTRNTAB {
 
 struct OP
 {
-    const(char)* str;   // opcode string
+    string str;   // opcode string
```
Comment (ljmf00) [inline src/dmd/backend/iasm.d:502]: I don't know if the header generator generates the proper struct for D arrays, but I can make it a union or use a `version` if this interferes with DMC. This is needed for CTFE to ease comparison of strings at compile-time.

Comment (maxhaton) [reply src/dmd/backend/iasm.d:502]: Header generator for what? DMC is written in D if you have C++ interop in mind

Comment (ljmf00) [reply src/dmd/backend/iasm.d:502]: > Header generator for what? DMC is written in D if you have C++ interop in mind

I don't know how walter interfaces this backend with DMC (https://github.com/DigitalMars/dmc). I don't think it is made in D but if this is a really old tree, then the change doesn't matter. If it does, I assumed the interfaces are not written by hand.

Comment (ljmf00) [reply src/dmd/backend/iasm.d:502]: Right, so nevermind. I guess all the extern(C++) thing and __gshared can go away then?

Comment (RazvanN7) [reply src/dmd/backend/iasm.d:502]: Yes.

**Comment by RazvanN7** [comment]
ping @ljmf00

**Comment by ljmf00** [comment]
I can't make this a compile-time check due to bootstrapping issues :/ I left a note in the code.

File: src/dmd/backend/ptrntab.d [hunk 2]
```diff
@@ -4799,1008 +4799,1014 @@ PTRNTAB2[2] aptb2SHA256MSG2 = /* SHA256MSG2 */ [
 //      The encoding is 0f C0+reg and it is always a 32
 //      bit operation
 
-immutable OP[] optab =
-[
+static immutable OP[] optab = [
```
Comment (dkorpel) [inline src/dmd/backend/ptrntab.d:4802]: What is `static` doing here?

Comment (ljmf00) [reply src/dmd/backend/ptrntab.d]: Nothing, it was a leftover from a lambda I used before. Should be good now.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**